homeassistant/components/config/script.py

```diff
@@ -2,13 +2,25 @@
 import asyncio
 
 from homeassistant.components.config import EditKeyBasedConfigView
-from homeassistant.components.script import SCRIPT_ENTRY_SCHEMA, async_reload
+from homeassistant.components.script import (
+    DOMAIN, SCRIPT_ENTRY_SCHEMA)
+from homeassistant.const import SERVICE_RELOAD
 import homeassistant.helpers.config_validation as cv
+from homeassistant.loader import bind_hass
 
 
 CONFIG_PATH = 'scripts.yaml'
 
 
+@bind_hass
+def async_reload(hass):
```

Comment:
This should be tagged with `@callback` as it calls an async method.

Reply:
Actually, from looking at the code, it should be an `async` function. The old code worked because it would actually schedule the async_call, but that seems weird.

Reply:
```python
async def async_reload(hass):
    await hass.services.async_call(DOMAIN, SERVICE_RELOAD)
```

Reply:
Just noticed it myself while looking at [config/group.py](https://github.com/home-assistant/home-assistant/blob/d1ad2cc22536df7ff65c578f054685a608077dc3/homeassistant/components/config/group.py#L15-L18)

Reply:
@balloob that is slow:
```python
def async_reload(hass):
   """This function return a coroutine."""
    return hass.services.async_call(DOMAIN, SERVICE_RELOAD)
```

?

Reply:
It's a bit slower but it gives proper stack traces. I don't want to do the old way anymore.